### ISSUE TITLE
classloader bug

### ISSUE THREAD

**Issue by snjeza** [projectlombok/lombok#3403]
A related issue - https://github.com/redhat-developer/vscode-java/issues/3046

The bug is caused by fixing #3373. It can't be reproduced using commit https://github.com/projectlombok/lombok/commit/d56b576c26d6e6a0621997a9758151cf41e7a74e

Steps To reproduce:
- install eclipse 4.27+
- install m2e
- import project [salut](https://github.com/redhat-developer/vscode-java/tree/master/test/resources/projects/maven/salut)
- close all editors
- restart eclipse
- call Project>Clean...
You will face the following exception;
```
!MESSAGE Errors running builder 'Java Builder' on project 'salut'.
!STACK 0
java.lang.NoClassDefFoundError: Could not initialize class lombok.launch.PatchFixesHider$ExtensionMethod
	at org.eclipse.jdt.internal.compiler.ast.MessageSend.resolveType(MessageSend.java:1003)
	at org.eclipse.jdt.internal.compiler.ast.UnaryExpression.resolveType(UnaryExpression.java:249)
	at org.eclipse.jdt.internal.compiler.ast.Expression.resolveTypeExpecting(Expression.java:1147)
	at org.eclipse.jdt.internal.compiler.ast.IfStatement.resolveIfStatement(IfStatement.java:288)
	at org.eclipse.jdt.internal.compiler.ast.IfStatement.resolve(IfStatement.java:317)
	at org.eclipse.jdt.internal.compiler.ast.AbstractMethodDeclaration.resolveStatements(AbstractMethodDeclaration.java:662)
	at org.eclipse.jdt.internal.compiler.ast.MethodDeclaration.resolveStatements(MethodDeclaration.java:388)
	at org.eclipse.jdt.internal.compiler.ast.AbstractMethodDeclaration.resolve(AbstractMethodDeclaration.java:571)
	at org.eclipse.jdt.internal.compiler.ast.TypeDeclaration.resolve(TypeDeclaration.java:1506)
	at org.eclipse.jdt.internal.compiler.ast.TypeDeclaration.resolve(TypeDeclaration.java:1631)
	at org.eclipse.jdt.internal.compiler.ast.CompilationUnitDeclaration.resolve(CompilationUnitDeclaration.java:668)
	at org.eclipse.jdt.internal.compiler.Compiler.process(Compiler.java:905)
	at org.eclipse.jdt.internal.compiler.ProcessTaskManager.run(ProcessTaskManager.java:145)
	at java.base/java.lang.Thread.run(Thread.java:833)
```

A workaround
- open a Java class
- restart eclipse
- call Project>Clean...

The issue happens when lombok.launch.PatchFixesHider.Transform.init(ClassLoader) is started from the org.eclipse.jdt.core.compiler.batch plugin,  See #3332

**Comment by Rawi01** [comment]
I already fixed that in #3378, I wasn't aware that this breaks the language server implementation. You can download a working version using the Github actions run artifact: https://github.com/projectlombok/lombok/actions/runs/4497732634

**Comment by snjeza** [comment]
It is fixed with https://github.com/projectlombok/lombok/pull/3378
